x21e8/liquid.py

```diff
@@ -1,71 +1,89 @@
+from x21e8.config import (
+    LQD_RPC_ENDPOINT_SCHEMA,
+    LQD_RPC_USER,
+    LQD_RPC_PASSWORD,
+    LQD_RPC_HOST,
+    LQD_RPC_PORT,
+    LIQUID_REGISTRATION_DOMAIN,
+)
+
 
 TOKEN_AMOUNT = 1
 VERSION = 0
-FEE_RATE = 0.03000000
+FEE_RATE = 0.00001
+
 
+def get_liquid_auth_proxy_url():
+    return f"{LQD_RPC_ENDPOINT_SCHEMA}://{LQD_RPC_USER}:{LQD_RPC_PASSWORD}@{LQD_RPC_HOST}:{LQD_RPC_PORT}"
 
-def issue_tokens(issue_request: IssuingRequest, nft_token, ipdl):
-    name = issue_request.name
-    ticker = issue_request.ticker
-    asset_amount = issue_request.amount
-    precision = issue_request.precision
 
+def get_keys(rpc_connection: AuthServiceProxy):
     pubkey = None
     asset_addr = None
     token_addr = None
-    rpc_connection = None
     try:
-        rpc_connection = AuthServiceProxy(
-            build_liquid_auth_proxy_url(),
-        )
-        new_addr = rpc_connection.getnewaddress("riddlemint", "legacy")
+        new_addr = rpc_connection.getnewaddress("riddlemint", "legacy")  # args: label, address type
         validate_addr = rpc_connection.getaddressinfo(new_addr)
         pubkey = validate_addr["pubkey"]
         asset_addr = new_addr
-        new_addr = rpc_connection.getnewaddress("MoteMagic", "legacy")
+        new_addr = rpc_connection.getnewaddress("MoteMagic", "legacy")  # args: label, address type
         token_addr = new_addr
-
     except JSONRPCException as json_exception:
         print("A JSON RPX exception occured: " + str(json_exception))
     except Exception as general_exception:
         print("An exception occured: " + str(general_exception))
 
-    print(pubkey)
-    print(asset_addr)
-    print(token_addr)
+    print(f"PubKey: {pubkey}")
+    print(f"Asset_addr(get_keys): {asset_addr}")
+    print(f"token_addr(get_keys): {token_addr}")
 
-    contract = f'{{"entity":{{"domain":"{LQD_RPC_ENDPOINT}"}}, "issuer_pubkey":"{pubkey}", "nft":{{"token":"{nft_token}", "ipld":"{ipdl}"}}, "name":"{name}", "precision":{precision}, "ticker":"{ticker}", "version":{VERSION}}}'
-    rpc_connection.settxfee(FEE_RATE)
-    print(contract)
-    contract_sorted = json.dumps(json.loads(contract), sort_keys=True, separators=(",", ":"))
+    return pubkey, asset_addr, token_addr
+
+
+def create_contract(issue_request: IssuingRequest, nft_token: str, cid: str, pubkey: str):
+    contract = f'{{"entity":{{"domain":"{LIQUID_REGISTRATION_DOMAIN}"}}, "issuer_pubkey":"{pubkey}", "nft":{{"token":"{nft_token}", "ipld":"{cid}"}}, "name":"{issue_request.name}", "precision":{issue_request.precision}, "ticker":"{issue_request.ticker}", "version":{VERSION}}}'
+    contract = json.loads(contract)
+    contract_sorted = json.dumps(contract, sort_keys=True, separators=(",", ":"))
     contract_hash = hashlib.sha256(six.ensure_binary(contract_sorted)).hexdigest()
-    print(contract_hash)
     contract_hash_rev = "".join(reversed([contract_hash[i : i + 2] for i in range(0, len(contract_hash), 2)]))
+    print(contract)
```

Comment:
do we want to print this information or do we want to add some logging at some point? :)